### DISCUSSION TITLE
Budibase October Release

### DISCUSSION THREAD

**Discussion by shogunpurple** [Budibase/budibase#3184] [Announcements]
# October Release

The October release is a huge step forward for the Budibase platform. 

We’ve been working hard, and we’ve listened to your much-appreciated feedback. Budibase will now be more dynamic, flexible, and powerful with the introduction of the full power of JavaScript to many places in the builder. 

## JS Scripting

You will now be able to write JavaScript almost anywhere in the builder, including mapping data, writing calculations, and performing operations on your bindings. 

When writing JavaScript in Budibase, you can view the results live in your design preview as soon as you save your code. 

Ultimately, this new functionality improves the flexibility of Budibase and allows you to build much more sophisticated applications.

![image](https://user-images.githubusercontent.com/11256663/139264922-f3881847-8b59-4cd7-b6f4-6cd506f13fba.png)

## Query Transformers

Query transformers introduce JavaScript to the Data section of Budibase - allowing you to execute a JavaScript expression on a response from a query on any of our data sources.

Map HTTP call responses into the shape you need, perform calculations using JavaScript on an array of results or perform filtering on your data using JavaScript snippets that you already know and love. 

![image](https://user-images.githubusercontent.com/11256663/139265036-ca1956cb-2fa5-4e6e-bd9b-614b2ee00648.png)

Read the documentation here: https://docs.budibase.com/building-apps/data/transformers

## No Access Role

You can now take advantage of the fact that Budibase provides fully featured Role Based Access Control out of the box, free and open source. The “No Access” role takes this security functionality further by defaulting newly invited users to have no access to any apps. These users will not even see the applications in their portal until access is granted to them by your budibase administrator.

You now have full control over what your users can see and access, and is particularly helpful if you are using Budibase within a team-based environment or organization.

## Bug fixes and Stability Improvements
* App export performance upgrades
* Fixes for filtering and querying data
* Ability to add databases with self-signed certificates
* Automations bug fixes and UX improvements
* Production -> development app replication improvements
* Much, much more.

**Comment by KernOil** [comment]
Clicking update from 0.9.168 says I already have the latest version. won't update to October release. 
![image](https://user-images.githubusercontent.com/80058238/139286314-569976a1-a1b7-4c5a-94b4-3e57baa0410e.png)

**Reply by tluyben** [reply]
Same here!

**Reply by shogunpurple** [reply]
According to docker hub, the latest tag has been pushed:

<img width="1262" alt="Screenshot 2021-10-28 at 17 40 58" src="https://user-images.githubusercontent.com/11256663/139289676-5d570ae1-64d6-4865-bccd-a9b6321fdfd2.png">

I was able to update one of our self hosted installations too to the latest.
<img width="595" alt="Screenshot 2021-10-28 at 17 42 30" src="https://user-images.githubusercontent.com/11256663/139289940-24666b5e-565a-442c-8aee-d794ce4a82b1.png">
 
 
 Could you make sure that your watchtower service is running?
 
 `docker ps` and you should see `containrrr/watchtower` there. If not, you could try the following command to update your installation.
 
 `docker-compose up --env-file hosting.properties up -d`

**Reply by KernOil** [reply]
@shogunpurple my watchtower docker container was off. turned it back on and now it is running the update. Thank you!
